src/components/NestedAccordionMenu/MenuItems/index.js

```diff
@@ -0,0 +1,98 @@
+import React, { Fragment, useState } from "react";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import ExpandLessIcon from "@material-ui/icons/ExpandLess";
+import Collapse from "@material-ui/core/Collapse";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import ListItemText from "@material-ui/core/ListItemText";
+import useStyles from "./styles";
+function MenuItem({ depth = 0, expanded, item, currentPath, ...rest }) {
+  const classes = useStyles({ depth: depth });
+  const [collapsed, setCollapsed] = useState(true);
+  const [highlightcolor, setHighlightcolor] = useState(false);
+  const { route, name, items, Icon } = item;
+  const mainroute = rest.mainroute;
+  let highlightclass = "itemNohighlighted";
+  function toggleCollapse() {
+    setCollapsed(prevValue => !prevValue);
+  }
+
+  function onClick(e) {
+    if (Array.isArray(items)) {
+      toggleCollapse();
+    }
+
+    if (currentPath === routeVal(mainroute, route)) {
+      if (!highlightcolor) {
+        setHighlightcolor(prevValue => !prevValue);
+      }
+    } else {
+      setHighlightcolor(prevValue => prevValue);
+    }
+  }
+  function routeVal(mainroute, route) {
+    return mainroute ? mainroute + route : route;
```

Comment:
Imho, each submenu should already have the correct url in the config object. It’s not necessary to nest them here.

Reply:
You mean that the child component url should be independent to the parent one?

Reply:
yes, nothing should change from config

Reply:
because I could want a sub menu item to have a external link for instance, then with that code it’s gonna break